client/client.go

```diff
@@ -488,7 +488,12 @@ func (c *simpleHTTPClient) Do(ctx context.Context, act httpAction) (*http.Respon
 		resp = rtresp.resp
 		switch {
 		case ctx.Err() != nil:
-			err = ctx.Err()
```

Comment:
@heyitsanthony this is so tricky with current vendored dependency.

So `ctx.Err() == context.Canceled` only true when the client vendors our client package by itself. So any project that doesn't do the vendoring will see the same error as the issue. If we do the string comparison like this patch, it will correctly returns the context error.

So strange, the Go types are both `*errors.errorString` and its error message are the same. Just the equal comparison operation fails when the client project is not using vendor.